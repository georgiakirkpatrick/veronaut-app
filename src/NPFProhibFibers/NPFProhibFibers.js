import React from 'react'
import formData from '../FORM_DATA'
import FormPage from '../FormPage/FormPage'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const noneOption = [
    {
        text: 'None of these',
        id: 100,
        checked: false
    }
]

const NPFProhibFibers = props => {
    const {
        currentPage,
        setPage,
        none,
        noneChange,
        pFiberChecks,
        pFiberChange,
        pageTurns
    } = props

    const nextButton = () => {
        const prohibFibs = []

        Object.keys(pFiberChecks).forEach(key => {
            if (pFiberChecks[key]) {
                prohibFibs.push(key)
            }
        })

        if (prohibFibs.length === 0 && !none[100]) {
            alert('Please select an option.')
        } else {
            setPage(currentPage + pageTurns)
        }
    }

    return (
        <div>
            <FormPage title='Fibers'>
                <FormPromptWithSub
                    prompt="Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread."
                    promptSubtitle="Check all that apply and click 'next'."
                />

                <FormCheckboxSection 
                    options={formData.prohibitedFibers}
                    selectedOptions={pFiberChecks}
                    handleChange={pFiberChange}
                />

                <FormCheckboxSection  
                    options={noneOption}
                    selectedOptions={none}
                    handleChange={noneChange}
                />
            </FormPage>

            <NPFFooter
                buttons='prevNext'
                previousButton={() => setPage(currentPage - pageTurns)} 
                nextButton={() => nextButton()}
            />
        </div>
    )
}

NPFProhibFibers.defaultProps = {
    currentPage: 1,
    handleChange: () => {},
    none: {},
    noneChange: () => {},
    pageTurns: 1,
    pFiberChange: () => {},
    pFiberChecks: {},
    selectedOptions: [],
    setPage: () => {},
    setPFiberChecks: () => {}
}

export default NPFProhibFibers