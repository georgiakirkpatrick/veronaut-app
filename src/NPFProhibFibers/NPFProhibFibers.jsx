import React from 'react'
import formData from '../FORM_DATA'
import FormPage from '../FormPage/FormPage'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const noneNotSure = [
  {
    text: 'None of these',
    id: 100,
    checked: false
  }
]

const options = formData.prohibitedFibers.slice(1, formData.prohibitedFibers.length)

const formattedProhib = options.sort((a, b) => 
  a.text > b.text ? 1 : -1
)

const prohibOptions = [
  ...formattedProhib,
  {
    text: 'Not sure',
    id: 101,
    checked: false
  }
] 

const NPFProhibFibers = props => {
  const {
    currentPage,
    setPage,
    none,
    noneChange,
    pFiberChecks,
    pFiberChange,
    pageTurns
  } = props

  const nextButton = () => {
    const prohibFibs = []

    Object.keys(pFiberChecks).forEach(key => {
      if (pFiberChecks[key]) {
        prohibFibs.push(key)
      }
    })

    if (prohibFibs.length === 0 && !none[100]) {
      alert('Please select an option.')
    } else {
      setPage(currentPage + pageTurns)
    }
  }

  return (
    <div>
      <FormPage title='Fibers'>
        <FormPromptWithSub
          prompt="Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread."
          promptSubtitle="Check all that apply and click 'next'."
        />

        <FormCheckboxSection 
          options={prohibOptions}
          selectedOptions={pFiberChecks}
          handleChange={pFiberChange}
        />

        <FormCheckboxSection  
          options={noneNotSure}
          selectedOptions={none}
          handleChange={noneChange}
        />
      </FormPage>

      <NPFFooter
        buttons='prevNext'
        previousButton={() => setPage(currentPage - pageTurns)} 
        nextButton={() => nextButton()}
      />
    </div>
  )
}

NPFProhibFibers.defaultProps = {
  currentPage: 1,
  handleChange: () => {},
  none: {},
  noneChange: () => {},
  pageTurns: 1,
  pFiberChange: () => {},
  pFiberChecks: {},
  selectedOptions: [],
  setPage: () => {},
  setPFiberChecks: () => {}
}

export default NPFProhibFibers