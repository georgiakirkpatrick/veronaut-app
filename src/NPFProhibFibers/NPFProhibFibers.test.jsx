import React from 'react'
import { render, screen } from '@testing-library/react'
import NPFProhibFibers from './NPFProhibFibers'
import { describe, it, expect } from 'vitest'

describe ('NPFProhibFibers', () => {
  it ('renders without crashing', () => {
    render(<NPFProhibFibers />)
    screen.debug()
  })

  it('renders the UI as expected', () => {
    const tree = render(<NPFProhibFibers />)
    expect(tree).toMatchSnapshot()
  })
})