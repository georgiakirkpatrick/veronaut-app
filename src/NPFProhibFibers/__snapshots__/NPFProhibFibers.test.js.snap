// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFProhibFibers renders the UI as expected 1`] = `
<div>
  <section
    className="FormPage"
  >
    <h1>
      Fibers
    </h1>
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          Check all that apply and click 'next'.
        </p>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <div
          className="empty"
        />
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={1}
              name={1}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={1}
            >
              Acetate
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={2}
              name={2}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={2}
            >
              Acrylic
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={11}
              name={11}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={11}
            >
              Elastane
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={13}
              name={13}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={13}
            >
              Lycra
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={3}
              name={3}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={3}
            >
              Microfiber
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={4}
              name={4}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={4}
            >
              Neoprene
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={5}
              name={5}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={5}
            >
              Nylon
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={6}
              name={6}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={6}
            >
              Olefin
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={7}
              name={7}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={7}
            >
              Polyamide
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={8}
              name={8}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={8}
            >
              Polyester
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={9}
              name={9}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={9}
            >
              Polyurethane
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={12}
              name={12}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={12}
            >
              Spandex
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={10}
              name={10}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={10}
            >
              Vinyl
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={101}
              name={101}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={101}
            >
              Not sure
            </label>
          </div>
        </div>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <div
          className="empty"
        />
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id={100}
              name={100}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor={100}
            >
              None of these
            </label>
          </div>
        </div>
      </div>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
